fog/view/server/src/fog_view_service.rs

```diff
@@ -188,4 +197,38 @@ impl<E: ViewEnclaveProxy, DB: RecoveryDb + Send + Sync> FogViewApi for FogViewSe
             send_result(ctx, sink, self.query_impl(request), logger)
         })
     }
+
+    /// Fulfills the query if the MultiViewStoreQueryRequest contains an
+    /// encrypted Query for the store. If it doesn't, then it responds with
+    /// an grpc error that contains the store's hostname.
+    fn multi_view_store_query(
+        &mut self,
+        ctx: RpcContext,
+        request: MultiViewStoreQueryRequest,
+        sink: UnarySink<MultiViewStoreQueryResponse>,
+    ) {
+        mc_common::logger::scoped_global_logger(&rpc_logger(&ctx, &self.logger), |logger| {
+            if let Err(err) = self.authenticator.authenticate_rpc(&ctx) {
+                return send_result(ctx, sink, err.into(), logger);
+            }
+            let mut response = MultiViewStoreQueryResponse::new();
+            for query in request.queries {
+                let result = self.query_impl(query);
+                if let Ok(attested_message) = result {
+                    response.set_query_response(attested_message);
```

Comment:
Since there are multiple queries in the request, shouldn't we include multiple responses?

Reply:
No. The `MultiViewStoreQueryRequest` contains a list of encrypted queries (of type `attest::Message`), which the Fog View Router encrypted for each Fog View Store that it knows about. It's the Fog View Store's (i.e. Fog View Server)  responsibility to look for the **one** encrypted query that it can decrypt and then fulfill that query by returning one query response. See the design [doc](https://docs.google.com/document/d/1-ju5efGK6vvuYjxj1pKhfFcpgGlornuxZYvOC-nsbjs/edit#heading=h.t7aeb1kbyq9h) for more info.